public/app/plugins/datasource/grafana/datasource.ts

```diff
@@ -8,24 +11,51 @@ import {
   isValidLiveChannelAddress,
   parseLiveChannelAddress,
   StreamingFrameOptions,
+  toDataFrame,
 } from '@grafana/data';
 
-import { GrafanaQuery, GrafanaAnnotationQuery, GrafanaAnnotationType, GrafanaQueryType } from './types';
-import { getBackendSrv, getGrafanaLiveSrv, getTemplateSrv, toDataQueryResponse } from '@grafana/runtime';
-import { Observable, of, merge } from 'rxjs';
-import { map, catchError } from 'rxjs/operators';
+import { GrafanaAnnotationQuery, GrafanaAnnotationType, GrafanaQuery, GrafanaQueryType } from './types';
+import AnnotationQueryEditor from './components/AnnotationQueryEditor';
+import { getDashboardSrv } from '../../../features/dashboard/services/DashboardSrv';
 
 let counter = 100;
 
 export class GrafanaDatasource extends DataSourceApi<GrafanaQuery> {
   constructor(instanceSettings: DataSourceInstanceSettings) {
     super(instanceSettings);
+    this.annotations = {
+      QueryEditor: AnnotationQueryEditor,
+      prepareAnnotation(json: any): AnnotationQuery<GrafanaAnnotationQuery> {
+        // Previously, these properties lived outside of target
+        // This should handle migrating them
+        json.target = json.target ?? {
+          type: json.type ?? GrafanaAnnotationType.Dashboard,
+          limit: json.limit ?? 100,
+          tags: json.tags ?? [],
+          matchAny: json.matchAny ?? false,
+        }; // using spread syntax caused an infinite loop in StandardAnnotationQueryEditor
+        return json;
+      },
+      prepareQuery(anno: AnnotationQuery<GrafanaAnnotationQuery>): GrafanaQuery {
```

Comment:
i'm not sure this is really a `GrafanaQuery`, but it doesn't seem to complain. this is basically `AnnotationQuery<GrafanaAnnotationQuery>` with a `queryType` and `refId`